### ISSUE TITLE
Suggestions for names for dotnet try

### ISSUE THREAD

**Issue by akshita31** [dotnet/try#226]
We are considering replacing the command "dotnet try" with something that is more informative of the intent.

**Comment by akshita31** [comment]
Current suggestions:
dotnet tutorial

**Comment by akshita31** [comment]
dotnet interactive

**Comment by loic-sharma** [comment]
I personally like `dotnet try`. It's short and sweet! Are customers confused by the current name?

**Comment by akshita31** [comment]
In alignment with the above discussion , we are looking to create a dotnet template which will include a csproj, a Program.cs with regions and a markdown file which has code snippets pointing to those regions.
This template would provide an easy way for the users to get started to create content that supports the interactive nature that comes with the dotnet try tooling.

Suggestions:
1. dotnet new tutorial

**Comment by akshita31** [comment]
@loic-sharma If we retain dotnet try, what do you suggest will be a good name for the template above. Till now we have been referring to it as "dotnet new try".

**Comment by jonsequitur** [comment]
@loic-sharma `dotnet new` ... `try`? Kind of weird. We were also looking at possibilities like this:

```console
> dotnet new tutorial       # <-- .NET Core console app
> dotnet new onlinetutorial # <-- Blazor-runnable, .NET Standard library
```

It might be helpful for the naming here to align with the naming of the tool itself.

**Comment by loic-sharma** [comment]
Hm. The template name is a good point, and I agree that the names should be aligned

I think `dotnet new tutorial` is a great name for the template! Instead of `dotnet new onlinetutorial`, would it be possible to accept a switch like `dotnet new tutorial --online`?

**Comment by LadyNaggaga** [comment]
I agree with @loic-sharma  `dotnet new tutorial --online` is clean and easy to  understand.  
I am worried that without using `try` we might mislead the user. 
`dotnet try new  tutorial   # <-- .NET Core console app` 
`dotnet try new tutorial --online   # <-- .NET Core console app`

**Comment by BillWagner** [comment]
I like "dotnet tutorial" and "dotnet interactive".  A couple other thoughts:

"dotnet explore"
"dotnet learn"

**Comment by colbylwilliams** [comment]
Is something like @LadyNaggaga's feasible?  This would also allow multiple templates for `try` in the future.

`dotnet try new core`
`dotnet try new blazor`
`dotnet try new bot-framework`

This could also facilitate future functionality similar to [CocoaPods](https://cocoapods.org/) [`pod try`](https://guides.cocoapods.org/terminal/commands.html#pod_try).  Where you could embed "tutorials" into your NuGet packages.

`dotnet try new package Newtonsoft.Json --version 12.0.2`

**Comment by KathleenDollard** [comment]
I really like the name 

```
dotnet try
```

It's a little quirky, which I think is desirable because this is not a tool that is constrained to any usage (like tutorials). 
(Of course, I prefered `dotnettry` or `trydotnet`, but I think that ship sailed.)

There are technical implications in some of the names for templates. Is the intent for this to be a global tool, templates in the current template system, or something else? 

If you are using the current template system, I think it would need to be:

```
dotnet new [something]
dotnet new try tutorial --online
```

If you aren't using the current template system, it would be good to understand why as I know they are working on improvements. 

If you're using this as an extension of your global tool, it seems logical to start with 

```
dotnet try
```

That's the landing place - the namespace. 

After that, you could do 

```
dotnet try new tutorial
```

It is most very unfortunate that this is a non-standard ordering and introduces is an order dependency on "new" depending on the technology of creation.  Weighing that against the confusion of a new verb is hard. Will it be a second template system? 

I do think you want to allow multiple starting points - multiple templates. The great thing it seems to be that the future possibilities are so broad and not yet imagined for trydotnet.

**Comment by rburkhal** [comment]
Because of the how things are done in the background, it appears to be a "live" instance of dotnet were the old constraints no longer apply. 

`dotnet live`

**Comment by jonsequitur** [comment]
@KathleenDollard We are working with the existing templating system: https://github.com/dotnet/try/pull/227.

**Comment by daveaglick** [comment]
+1 from me for keeping `dotnet try` and using `dotnet try new ...` for templates.

That said, if I _had_ to pick another name, some thoughts are:
- `dotnet lab` (like it almost as much as `try` - short and embodies both learning and experimentation)
- `dotnet learn` (mentioned above, like it too)
- `dotnet play` (as in “to play with this code”)

**Comment by colombod** [comment]
I am still on `dotnet new try` for templates.

**Comment by melgharbawy** [comment]
Considering the MS Learn is one of the go to places for learning... would it be a bad idea to consider replacing the “new” keyword with “learn” while opening the possibility of introducing almost identical templates to those that already come out of the box... examples would be:
dotnet learn web
dotnet learn webapi 
dotnet learn console
... etc

Potentially even adding the flexibility of enabling the same options or similar to those that exist with new’ing up new projects?

**Comment by GalaxiaGuy** [comment]
If the name became "dotnet explore", the template could be `dotnet new exploration`.

**Comment by dprice809** [comment]
I think try is a great name, but I find that google searches clash a lot with try/catch documentation and such.
